staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -163,6 +166,7 @@ var (
 
 	warningNoLastAppliedConfigAnnotation = "Warning: resource %[1]s is missing the %[2]s annotation which is required by %[3]s apply. %[3]s apply should only be used on resources created declaratively by either %[3]s create --save-config or %[3]s apply. The missing annotation will be patched automatically.\n"
 	warningChangesOnDeletingResource     = "Warning: Detected changes to resource %[1]s which is currently being deleted.\n"
+	warningMigrationPatchFailed          = "Warning: attempted migration of client-side-apply managed fields to server-side-apply failed. Please re-apply to try again. Error: %[1]s\n"
```

Comment:
We're definitely going to have people wonder what's going on when they read that error. I'm trying to see what I would think and how would I debug given such an error:
1. Have I requested that migration? Why does it think it should migrate?
2. What is it migrating and how?

That's even more true since most of this migration is done behind-the-scene without letting users know what's going on.  Is there anything we can do to explain what's going on?

Reply:
I would make the warnings a little more contextual (based on what's actually going on), they show the error but since the same string is used in multiple location, it might be hard to investigate, even for us.